### PR TITLE
feat: enhance bridge quote handling for hardware wallets cp-13.24.0

### PR DESCRIPTION
**Pull request by dawnseeker8**
## **Description**

Hardware wallets (Ledger, QR-based) cannot participate in gasless flows — neither EIP-7702 delegation nor STX `sendBundle` — because both mechanisms require capabilities that HW devices don't support. This PR fixes a regression where the `gasIncluded` (STX) flag was sent ungated for HW wallets in the quote request, and adds Monad-specific native reserve validation for 7702 sponsored transactions.

### Problem

1. **`gasIncluded` (STX path) regression** — A prior refactor correctly identified that `useGasIncluded7702` already gates HW wallets for the 7702 path, so the extra `!isUsingHardwareWallet` wrapper on `gasIncluded7702` was redundant. However, the same reasoning was incorrectly applied to `gasIncluded` (STX path). Unlike 7702, `getIsGasIncluded` does **not** check for HW wallets, and `useGaslessSupportedSmartTransactions` in the confirmations flow explicitly excludes HW wallets (`!isHardwareWalletAccount && isSmartTransaction && sendBundleSupported`). This created a mismatch: quotes were fetched with gas included, but execution would refuse to sponsor gas.

2. **Monad native reserve** — On Monad, 7702 sponsored transactions require the sender to hold `≥ 10 MON` native balance as a relay requirement. Without this check, users could attempt a gasless swap that drains their balance below this threshold, causing the relay to reject the tx on-chain.

### Solution

**Request-time gating (matching the [mobile pattern](https://github.com/MetaMask/metamask-mobile/pull/27615)):**
- `useGasIncluded7702` — already returns `false` for HW wallets (7702 path gated at source)
- `effectiveGasIncluded = !isUsingHardwareWallet && gasIncluded` — restored in `prepare-bridge-page.tsx` to gate the STX path at quote-request time
- Quote request sends `gasIncluded: false` and `gasIncluded7702: false` for HW wallets, so the backend returns correct (non-sponsored) quotes directly — no post-quote normalization needed

**Defense-in-depth:**
- `_getBaseValidationErrors` selector: `isGasless` includes `!isHardwareWalletAccount` gate
- Quote card UI: `gasIncluded` rendering gated with `!isHardwareWalletAccount`; network-level sponsorship path gated with `!isHardwareWalletAccount`

**Monad reserve validation:**
- Added `isMonad7702` check in `_getBaseValidationErrors` that enforces `≥ 10 MON` native reserve when `gasIncluded7702` is active on Monad
- Surfaces `isInsufficientGasBalance` / `isInsufficientGasForQuote` when the reserve is not met

**Minor fixes:**
- `activeQuote ?? null` in `useQuoteFetchEvents` and `prepare-bridge-page` for type safety

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/41115?quickstart=1)

## **Changelog**

CHANGELOG entry: Fixed gas sponsorship display for hardware wallet accounts on gasless networks (Monad, Sei) — hardware wallets now correctly fall back to user-paid gas fees.

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/browse/NEB-767

### Hardware wallet + gasless network (Monad/Sei)
1. Connect a Ledger or QR-based hardware wallet and set it as the active account
2. Navigate to the Bridge/Swap page
3. Select a gasless network (e.g. Monad) as the source chain
4. Enter a swap amount and wait for quotes
5. Verify **no** "Gas fees sponsored" or "Gas fees included" labels appear — the user should see standard network fee display
6. Confirm the quote request is sent with `gasIncluded: false` and `gasIncluded7702: false` (check network tab)

### Non-HW wallet + gasless network (control)
7. Switch to a regular (non-hardware) EOA account
8. Repeat steps 2–4 on Monad
9. Verify "Gas fees sponsored" or "Gas fees included" labels **do** appear as expected

### Monad native reserve (10 MON minimum)
10. With a non-HW account on Monad holding ~15 MON
11. Attempt a swap for 10 MON (leaving 5 MON — below the 10 MON reserve)
12. Verify an insufficient gas balance warning is shown
13. Reduce swap to 3 MON (leaving 12 MON — above reserve)
14. Verify the warning clears and the swap proceeds normally

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> **Medium Risk**
> Touches bridge quote validation and fee-display logic, which can affect whether users can submit swaps and what gas warnings/UI they see. Risk is mitigated by added unit tests covering HW-wallet and Monad reserve edge cases.
> 
> **Overview**
> Improves gasless-quote handling for hardware wallets by **gating sponsorship-only flows** (EIP-7702/`gasSponsored`) while still allowing `gasIncluded` (STX) quotes, and updates the quote card to never show sponsored-fee UI for HW accounts.
> 
> Adds **Monad-specific validation** that requires a `>= 10 MON` native reserve when `quoteRequest.gasIncluded7702` is used, surfacing `isInsufficientGasBalance` when the post-trade balance would drop below the relay threshold.
> 
> Tightens type-safety by consistently passing `activeQuote ?? null` into helpers, and adds targeted tests for the new HW gating, Monad reserve checks, and quote-fetch analytics emission.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit c4c5bd95663e05efae62736177baaf379025ac16. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: ui/ducks/bridge/selectors.ts [hunk 1]
```diff
@@ -746,9 +746,14 @@ const _getBaseValidationErrors = createDeepEqualSelector(
...
       activeQuote?.quote ?? {};
+    // gasIncluded7702 and gasSponsored are gated at request time via
+    // useGasIncluded7702 (returns false for HW), so the backend won't
+    // return those flags for HW accounts.
+    // gasIncluded (STX path) is gated at request time in
+    // prepare-bridge-page, but we also gate here as defense-in-depth.
     const isGasless =
       !isHardwareWalletAccount &&
-      (gasIncluded7702 || gasIncluded || gasSponsored);
+      Boolean(gasIncluded7702 || gasIncluded || gasSponsored);
```
Comment (micaelae) [inline ui/ducks/bridge/selectors.ts]: HW wallet should also only gate 7702 and sponsored here

```suggestion
    // prepare-bridge-page, but we also gate here as defense-in-depth.
    const isGasless = isHardwareWalletAccount
      ? gasIncluded
      : gasIncluded || gasIncluded7702 || gasSponsored;
```

File: ui/pages/bridge/prepare/prepare-bridge-page.tsx [hunk 2]
```diff
@@ -174,8 +174,11 @@ const PrepareBridgePage = ({
   const keyring = useSelector(getCurrentKeyring);
   const isUsingHardwareWallet = isHardwareKeyring(keyring?.type);
 
+  // Defense-in-depth: useGasIncluded7702 already returns false for HW
+  // wallets, and getIsGasIncluded (STX path) is gated elsewhere, but we
+  // guard both here to protect against future hook/selector changes.
   const effectiveGasIncluded = !isUsingHardwareWallet && gasIncluded;
```
Comment (micaelae) [inline ui/pages/bridge/prepare/prepare-bridge-page.tsx]: gasIncluded quotes can be submitted by hardware wallets. only gasIncluded7702 and gasSponsored quotes need to be excluded

```suggestion
  const effectiveGasIncluded = gasIncluded;
```

File: ui/ducks/bridge/selectors.ts [hunk 3]
```diff
@@ -757,6 +762,16 @@ const _getBaseValidationErrors = createDeepEqualSelector(
...
 
+    // Monad requires >= 10 MON native reserve for 7702 sponsored txs.
+    // Without this balance the relay rejects the tx on-chain.
+    const MONAD_MIN_RESERVE = '10';
+    const srcHexChainId = srcChainId
+      ? getMaybeHexChainId(String(srcChainId))
+      : undefined;
+    const isMonad7702 =
+      srcHexChainId === CHAIN_IDS.MONAD &&
+      Boolean(gasIncluded7702 || quoteRequest.gasIncluded7702);
```
Comment (micaelae) [inline ui/ducks/bridge/selectors.ts]: Instead of modifying the conditions below, I suggest reusing the 'minimumBalanceToUse' logic for the Monad reserve amount. Doing so removes the need for any other changes in this selector

```suggestion
      quoteRequest.srcChainId ?? activeQuote?.quote?.srcChainId;
    let minimumBalanceToUse =
      srcChainId && isSolanaChainId(srcChainId)
        ? minimumBalanceForRentExemptionInSOL
        : '0';

    // Monad requires >= 10 MON native reserve for 7702 sponsored txs.
    // Without this balance the relay rejects the tx on-chain.
    const MONAD_MIN_RESERVE = '10';
    const srcHexChainId = srcChainId
      ? getMaybeHexChainId(String(srcChainId))
      : undefined;

    if (srcHexChainId === CHAIN_IDS.MONAD) {
      minimumBalanceToUse = MONAD_MIN_RESERVE;
    }
```

**Review by micaelae — APPROVED** [review]

**Review by SteP-n-s — APPROVED** [review]
